Sources/Parser/Parser.swift

```diff
@@ -472,12 +455,8 @@ extension Parser {
   func parseContractBehaviorDeclaration() throws -> ContractBehaviorDeclaration {
     let contractIdentifier = try parseIdentifier()
 
-    let typeStates: [TypeState]?
-    if let typeStateGroup = try? parseTypeStateGroup() {
-      typeStates = typeStateGroup.typeStates
-    } else {
-      typeStates = nil
-    }
+    let _ = attempt(try consume(.punctuation(.at)))
```

Comment:
I don't think this change is right - this would make the @ optional for type state groups. What was the issue with consuming the `@` inside `parseTypeStateGroup`?

Reply:
parseTypeStateGroup is being reused in the contract declaration i.e. contract Name (State1, State2, State3) where I've treated (State1, State2, State3) as a type state group.

Reply:
We could push that inside and make a new function for parsing type state declaration groups, but that seems a little excessive